soc/intel/intel_adsp/ace/asm_memory_management.h

```diff
@@ -79,5 +81,27 @@
 	addi \au, \au, -1
 	bnez \au, 2b
 .endm
+
+.macro m_ace_hpsram_power_down_entire ax, ay, az, au
+	/* Read the HPSRAM bank count from ACE_L2MCAP register */
+	movi \au, DT_REG_ADDR(DT_NODELABEL(hsbcap))
+	l32i \au, \au, 0
+	extui \au, \au, 0, 8 /* Bank count is in the lower 8 bits */
+
+	movi \ay, 1             /* Power down command */
+
+	/* Calculate the address of the HSxPGCTL register */
+	movi \az, DT_REG_ADDR(DT_NODELABEL(hsbpm))
```

Comment:
Since we discussed the use of `movi` before, do I understand it correctly, that up to this point `movi`s are good, but not inside the loop, where memory banks are already being switched off? Maybe a follow-up PR to add a comment here about that